crates/auth/src/lib.rs

```diff
@@ -1,2 +1,23 @@
 // SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
 // SPDX-License-Identifier: Apache-2.0
+mod error;
+mod providers;
+mod redirect;
+mod session;
+
+use axum::{extract::Extension, routing::get, Router};
+use providers::github;
+
+pub use providers::Provider;
+pub use session::{Session, COOKIE_NAME};
+
+pub fn app(host: &str, client_id: String, client_secret: String) -> Router {
```

Comment:
It's not clear what the parameters are, we probably want an `AppBuilder`, which would have a `github` method taking these and enable github auth. (can be a follow-up TODO issue, but then we have to rename the parameters or introduce some kind of `GithubConfig`